compiler/src/iree/compiler/Codegen/Common/GPU/GPUHeuristics.cpp

```diff
@@ -277,16 +277,20 @@ static LogicalResult canTargetIntrinsic(const GPUMatmulShapeType &problem,
 
 static SmallVector<int64_t>
 getBestKTileSizes(const GPUMatmulShapeType &problem,
-                  const GPUMatmulShapeType &intrinsic,
+                  const GPUIntrinsicType &intrinsic,
                   const GPUMMAHeuristicSeeds &seeds) {
   // kTotalTileCounts is similar to m/nTotalTileCounts, representing the total
   // number of intrinsics along the K dimensions needed to fill the problem.
   // For the problem described above {M:[4, 16], N:[2, 32], K[3, 128]} with a
   // 16x16x16 intrinsic, then:
   //  - kTotalTileCounts would be 3 * (128/16) = 24
   SmallVector<int64_t, 2> kTotalTileCounts = problem.kSizes;
-  kTotalTileCounts.back() =
-      llvm::divideCeil(problem.kSizes.back(), intrinsic.kSizes[0]);
+  for (auto [kTotalTileCount, intrinsicKSize] : llvm::zip(
+           MutableArrayRef{kTotalTileCounts}.take_back(intrinsic.kSizes.size()),
+           intrinsic.kSizes)) {
```

Comment:
Use zip_equal for ranges of equal length
```suggestion
  for (auto [kTotalTileCount, intrinsicKSize] : llvm::zip_equal(
           MutableArrayRef{kTotalTileCounts}.take_back(intrinsic.kSizes.size()),
           intrinsic.kSizes)) {
```

Reply:
done!